File: backend/golang/middleware/cors.go
package middleware

import (
	"net/http"

	"github.com/gorilla/handlers"
)

// CORS 配置CORS中间件
func CORS() func(http.Handler) http.Handler {
	return handlers.CORS(
		handlers.AllowedOrigins([]string{"*"}),
		handlers.AllowedMethods([]string{"GET", "POST", "PUT", "DELETE", "OPTIONS"}),
		handlers.AllowedHeaders([]string{"Content-Type", "Authorization"}),
	)
}

// CORSWithConfig 可配置的CORS中间件
func CORSWithConfig(origins []string, methods []string, headers []string) func(http.Handler) http.Handler {
	return handlers.CORS(
		handlers.AllowedOrigins(origins),
		handlers.AllowedMethods(methods),
		handlers.AllowedHeaders(headers),
	)
}

File: backend/hono/src/middleware/cors.ts
/**
 * CORS中间件
 */

import { cors } from 'hono/cors';
import { settings } from '../config';

export const corsMiddleware = cors({
  origin: (origin) => {
    if (settings.cors.allowedOrigins.includes('*')) {
      return origin;
    }
    return settings.cors.allowedOrigins.includes(origin || '') ? origin : null;
  },
  allowMethods: ['GET', 'POST', 'PUT', 'DELETE', 'OPTIONS'],
  allowHeaders: ['Content-Type', 'Authorization', 'X-Requested-With'],
  maxAge: 86400,
}); 